Chapter6/super_process.erl

```diff
@@ -0,0 +1,107 @@
+-module(super_process).
+-export([start_link/2, stop/1, add_child/2, stop_child/2]).
+-export([init/1]).
+
+% Interface
+start_link(Name, ChildSpecList) ->
+	register(Name, spawn_link(super_process, init, [ChildSpecList])), ok.
+
+stop(Name) ->
+	Name ! {stop, self()},
+	receive {reply, Reply} -> Reply end.
+
+
+% Exercise 6-3: A Supervisor Process (start_child & stop_child)
+add_child(Name, Process) ->
+	Name ! {add, self(), Process},
+	receive {reply, Reply} -> Reply end.
+
+stop_child(Name, Pid) ->
+	Name ! {remove, self(), Pid},
+	receive {reply, Reply} -> Reply end.
+% Interface
+
+
+init(ChildSpecList) ->
+	process_flag(trap_exit, true),
+	loop(start_children(ChildSpecList)).
+
+start_children([]) -> [];
+start_children([{M,F,A,T} | ChildSpecList]) ->
+	case ( catch apply(M,F,A) ) of
+		{ok, Pid} ->
+			[ {Pid, {M,F,A,T,{0,0}}} | start_children(ChildSpecList) ];
+		_ ->
+			start_children(ChildSpecList)
+	end.
+
+
+% Don't know how to make the module not available to test 6-3 part two (spawn a child whose module is not available?)
+
+restart_child(Pid, ChildList) ->
+	{value, {Pid, {M,F,A,T,R}}} = lists:keysearch(Pid, 1, ChildList),
+
+	{Min, Ammount} = R,
+	{_,{_,CurrentMin,_}} = calendar:now_to_local_time(erlang:timestamp()),
```

Comment:
it looks like if process terminates each hour at the same minute, then it'll be remove when threshold is reached